lib/file.c

```diff
@@ -400,6 +406,64 @@ static CURLcode file_upload(struct Curl_easy *data)
   return result;
 }
 
+/*
+ * directory_list() will read all entries in a directory and prepare
+ * a temporary char * buffer with the full list, with entries separated
+ * by \n, for file_do().
+ */
+static CURLcode directory_list(struct FILEPROTO *file, char **data, curl_off_t *size) {
+  CURLcode result = CURLE_OK;
+  DIR *dir;
+  struct dirent *entry;
+  curl_off_t entry_len;
+
+  *data = NULL;
+  *size = 0;
+
+  if(!file) {
```

Comment:
can `file` actually and legitimately be NULL here? Shouldn't this rather be an assert since it should never happen in real life?